packages/core-mobile/app/services/ledger/LedgerService.ts

```diff
@@ -0,0 +1,693 @@
+            vmType: NetworkVMType.EVM
+          }).replace('/0/0', ''),
+          key: evmXpubResponse.publicKey.toString('hex'),
+          chainCode: evmXpubResponse.chain_code.toString('hex')
+        },
+        avalanche: {
+          path: getAddressDerivationPath({
+            accountIndex: 0,
+            vmType: NetworkVMType.AVM
+          }).replace('/0/0', ''),
+          key: avalancheXpubResponse.publicKey.toString('hex'),
+          chainCode: avalancheXpubResponse.chain_code.toString('hex')
+        }
+      }
+    } catch (error) {
+      // Provide more specific error messages
+      if (error instanceof Error) {
+        if (error.message.includes('app not initialized')) {
+          throw new Error(
+            'Avalanche app not open on device. Please open the Avalanche app and try again.'
+          )
+        } else if (error.message.includes('device not found')) {
+          throw new Error(
+            'Ledger device not found. Please ensure your device is connected and unlocked.'
+          )
+        } else if (error.message.includes('returnCode')) {
+          throw new Error(
+            'Device returned an error. Please ensure the Avalanche app is open and ready.'
+          )
+        } else {
+          throw new Error(
+            `Failed to get extended public keys: ${error.message}`
+          )
+        }
+      }
+      Logger.error('=== getExtendedPublicKeys FAILED ===', error)
+      throw new Error(`Failed to get extended public keys: ${error}`)
+    }
+  }
+
+  // Check if Solana app is open
+  async checkSolanaApp(): Promise<boolean> {
+    if (!this.#transport) {
+      return false
+    }
+
+    try {
+      // Create fresh Solana app instance
+      const solanaApp = new AppSolana(this.transport)
+      // Try to get a simple address to check if app is open
+      // Use a standard Solana derivation path
+      const testPath = "m/44'/501'/0'"
+      await solanaApp.getAddress(testPath, false)
+      return true
+    } catch (error) {
+      Logger.error('Solana app not open or not available', error)
+      return false
+    }
+  }
+
+  // Get Solana address for a specific derivation path
+  async getSolanaAddress(
+    derivationPath: string
+  ): Promise<{ address: Buffer; publicKey: Buffer }> {
+    await this.waitForApp(LedgerAppType.SOLANA)
+    const solanaApp = new AppSolana(this.transport)
+    return await solanaApp.getAddress(derivationPath, false)
+  }
+
+  // Get Solana public keys using SDK function (like extension)
+  async getSolanaPublicKeys(
+    startIndex: number,
+    count: number
+  ): Promise<PublicKeyInfo[]> {
+    // Create a fresh AppSolana instance for each call (like the SDK does)
+    const freshSolanaApp = new AppSolana(this.transport)
+    const publicKeys: PublicKeyInfo[] = []
+
+    try {
+      for (let i = startIndex; i < startIndex + count; i++) {
+        // Use correct Solana derivation path format
+        const derivationPath = `44'/501'/0'/0'/${i}`
```

Comment:
let's create a util function to generate the full path and a const for the path prefix